File: split/pauseforinstructions.pas
procedure pauseforinstructions;
begin
  if OKtointerrupt then
  begin
    interaction := 3;
    if (selector = 18) or (selector = 16) then
      selector := selector + 1;
    begin
      if interaction = 3 then;
      printnl(262);
      print(296);
    end;
    begin
      helpptr := 3;
      helpline[2] := 297;
      helpline[1] := 298;
      helpline[0] := 299;
    end;
    deletionsallowed := false;
    error;
    deletionsallowed := true;
    interrupt := 0;
  end;
end;

File: cpp/pauseforinstructions.cpp
#include "pauseforinstructions.h"
#include "printnl.h"
#include "print.h"
#include "error.h"
#include "texte.h"

void pauseforinstructions(void)
{
	if (OKtointerrupt)
	{
		interaction = error_stop_mode;
		if (selector == log_only || selector == no_print)
			selector++;
		printnl("! ");
		print("Interruption");
		helpptr = 3;
		helpline[2] = "You rang?";
		helpline[1] = "Try to insert some instructions for me (e.g.,`I\\showlists'),";
		helpline[0] = "unless you just want to quit by typing `X'.";
		deletionsallowed = false;
		error();
		deletionsallowed = true;
		interrupt = 0;
	}
}
